bigquery/google/cloud/bigquery/dataset.py

```diff
@@ -283,28 +293,38 @@ def from_string(cls, dataset_id, default_project=None):
             >>> DatasetReference.from_string('my-project-id.some_dataset')
             DatasetReference('my-project-id', 'some_dataset')
 
+            >>> DatasetReference.from_string('prefix:my-project-id.some_dataset')
+            DatasetReference('my-project-id', 'some_dataset')
+
         Raises:
             ValueError:
                 If ``dataset_id`` is not a fully-qualified dataset ID in
                 standard SQL format.
         """
         output_dataset_id = dataset_id
         output_project_id = default_project
-        parts = dataset_id.split(".")
+        with_prefix = _W_PREFIX.match(dataset_id)
+        if with_prefix is None:
+            parts = dataset_id.split(".")
+        else:
+            parts = with_prefix.group("ref").split(".")
 
         if len(parts) == 1 and not default_project:
             raise ValueError(
                 "When default_project is not set, dataset_id must be a "
-                "fully-qualified dataset ID in standard SQL format. "
-                'e.g. "project.dataset_id", got {}'.format(dataset_id)
+                "fully-qualified dataset ID in standard SQL format, "
```

Comment:
Ditto. "prefix" is both rare and confusing. Please revert the error message changes.